novawallet/Modules/Swaps/Setup/SwapSetupPresenter.swift

```diff
@@ -134,21 +134,29 @@ final class SwapSetupPresenter {
         view?.didReceiveAmount(receiveInputViewModel: amountInputViewModel)
     }
 
-    private func absoluteValue(for input: AmountInputResult?) -> Decimal? {
+    private func getPayAmount(for input: AmountInputResult?) -> Decimal? {
         guard
             let input = input,
             let payChainAsset = payChainAsset else {
             return nil
         }
-        guard let transferrableBalanceDecimal =
-            Decimal.fromSubstrateAmount(
-                assetBalance?.transferable ?? 0,
-                precision: payChainAsset.asset.displayInfo.assetPrecision
-            ) else {
-            return nil
+        let includedFee: BigUInt
+        switch input {
+        case .rate:
+            includedFee = fee ?? 0
+        case .absolute:
+            includedFee = 0
         }
 
-        return input.absoluteValue(from: transferrableBalanceDecimal)
+        let transferable = assetBalance?.transferable ?? 0
+        let balance = max(transferable - includedFee, 0)
```

Comment:
I think the difference may overflow and crash the app. It is better to compare BigUInts.